src/encoding/plain_byte_array/decoder.rs

```diff
@@ -0,0 +1,37 @@
+use crate::encoding::get_length;
+
+/// Decodes according to [Plain strings](https://github.com/apache/parquet-format/blob/master/Encodings.md#plain-plain--0),
+/// prefixes, lengths and values
+/// # Implementation
+/// This struct does not allocate on the heap.
+#[derive(Debug)]
+pub struct Decoder<'a> {
+    values: &'a [u8],
+    index: usize,
```

Comment:
I think that it is possible to remove this `index` by re-writing `self.values`. In `next`, something like

```rust
result = Some(&self.values[4..4 + next_len]);
self.values = &values[4 + next_len..];
result
```

together with the condition `if self.values.is_empty()`

This is also how `ExactChunk` is implemented in the `std`